src/journal/journald-server.c

```diff
@@ -846,7 +847,10 @@ static void dispatch_message_real(
                 }
 #endif
 
-                r = cg_pid_get_path_shifted(ucred->pid, s->cgroup_root, &c);
+                if (cgroup)
+                        c = cgroup, r = 0;
```

Comment:
This comma syntax, while legal C, is a bit rare and misleading IMHO. Perhaps move the `if (r >= 0)` into the `else` branch instead?